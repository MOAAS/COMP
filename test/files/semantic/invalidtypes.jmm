import io.println(int) void;
import Quicksort;

//class Simple extends Simple {
class Simple extends Quicksort {
    Simple a;
    Quicksort b;
    invalidfieldtype c;

    public static void main(String[] arg){
        Simple a;
        Quicksort b;
        quicksort c;
        simple d;
        Int e;
        Boolean f;

        int g;
        int[] h;
        boolean i;
    }

    public invalidreturntype fun(invalidparam1 arg1, invalidparam2 arg2) {
        return a;
    }

    public invalidreturntype2 fun(invalidparam3 arg1, invalidparam4 arg2) {
        return a;
    }
 }