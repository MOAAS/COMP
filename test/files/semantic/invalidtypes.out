Error: Invalid variable type: "invalidfieldtype" at line 8, column 5.
Error: Invalid variable type: "quicksort" at line 13, column 9.
Error: Invalid variable type: "simple" at line 14, column 9.
Error: Invalid variable type: "Int" at line 15, column 9.
Error: Invalid variable type: "Boolean" at line 16, column 9.
Error: Invalid variable type: "invalidparam1" at line 23, column 34.
Error: Invalid variable type: "invalidparam2" at line 23, column 54.
Error: Invalid variable type: "invalidreturntype" at line 23, column 12.
Error: Invalid variable type: "invalidparam3" at line 27, column 35.
Error: Invalid variable type: "invalidparam4" at line 27, column 55.
Error: Invalid variable type: "invalidreturntype2" at line 27, column 12.
